daemon/tests/happy_path.rs

```diff
@@ -86,28 +71,34 @@ async fn taker_takes_order_and_maker_rejects() {
     assert!(matches!(maker_cfd.state, CfdState::Rejected { .. }));
 }
 
+// Helper function setting up a "happy path" wallet mock
+fn mock_wallet_transaction(wallet: &mut MutexGuard<'_, harness::mocks::wallet::MockWallet>) {
+    #[allow(clippy::redundant_closure)] // clippy is in the wrong here
+    wallet
+        .expect_build_party_params()
+        .times(1)
+        .returning(|msg| build_party_params(msg));
+
+    wallet
+        .expect_sign()
+        .times(1)
+        .returning(|| Ok(dummy_partially_signed_transaction()));
+    wallet
+        .expect_broadcast()
+        .times(1)
+        .returning(|| Ok(dummy_tx_id()));
+}
+
 #[tokio::test]
 async fn taker_takes_order_and_maker_accepts_and_contract_setup() {
     let _guard = init_tracing();
-    let (mut maker, mut taker) = start_both(
-        Oracle::default()
-            .with_dummy_announcement(harness::cfd_protocol::OliviaData::example_0().announcement()),
-        Monitor,
-        Wallet::default()
-            .with_wallet_info()
-            .with_party_params()
-            .with_psbt(harness::bdk::dummy_partially_signed_transaction())
-            .with_txid(harness::bdk::dummy_tx_id()),
-        Oracle::default()
-            .with_dummy_announcement(harness::cfd_protocol::OliviaData::example_0().announcement()),
-        Monitor,
-        Wallet::default()
-            .with_wallet_info()
-            .with_party_params()
-            .with_psbt(harness::bdk::dummy_partially_signed_transaction())
-            .with_txid(harness::bdk::dummy_tx_id()),
-    )
-    .await;
+    let (mut maker, mut taker) = start_both().await;
+
+    mock_wallet_transaction(&mut maker.mocks.wallet().await);
```

Comment:
I am not sure this should be done in the beginning;  do you follow some kind of guide here? 
I would have always instructed what is expected directly before we need it, so I would move this code together with the expected oracle event right before the maker accepts.

Reply:
from the point of the mock, it does not matter. it can be instructed arbitrarily early (it's "when this method gets called, return the following value ").

Happy to move it if it makes it more readable for others (arguably we're move specific then that we never get these calls before we expect them).

Reply:
moved these around: build params is just before the contract setup, and sign & broadcast mock after contract setup.